test/mingap_test.cpp

```diff
@@ -80,18 +80,32 @@ TEST(MinGap, SphereSphereOutOfBounds) {
 
 TEST(MinGap, ClosestPointOnEdge) {
   auto a = Manifold::Cube();
-  auto b = Manifold::Sphere(1.0f).Translate({3.0f, 0.0f, 0.5f});
+  auto b = Manifold::Cube().Translate({2.0f, 1.0f, 0.5f});
 
   float distance = a.MinGap(b, 10);
 
-  EXPECT_FLOAT_EQ(distance, 1);
+  EXPECT_FLOAT_EQ(distance, 1.0f);
```

Comment:
Might be good to cut all the search distances down to say, 10% more than the expected distance, to tighten the tests regarding BVH culling.

Reply:
Ah, good idea. I cut them down and looks like it's not working correctly, still need to figure out why though.

Reply:
@elalish @pca006132 
Can you check out [these](https://github.com/elalish/manifold/pull/765/files#diff-fb49545a59b0ea198e1005fcf8f5390dffff8f1823090d521b1abcd85a8cbb3eR102-R127) two test cases?

For this more complex test case, if I put in a search distance 10% more than the actual distance, the method returns the search distance, even though it's larger than the actual distance. 

However, If I put in a dummy boolean operation before calling MinGap, it seems to work correctly. I suspect something might be wrong with my sorting? What could these boolean ops be doing that make it work?

Reply:
The BVH paper I implemented is here: https://github.com/elalish/manifold/blob/master/docs/ParallelBVH.pdf

Reply:
Seems like sorting the other manifold's boxes were causing the issue. If I leave them as is and only sort the original one it works.